File: callers/android/src/main/kotlin/DefaultConnectionHandler.kt
import java.io.InputStream
import java.io.OutputStream
import java.net.Socket

/**
 * Базовая реализация ConnectionHandler для тестирования.
 * Использует стандартное TCP соединение с целевым сервером.
 */
class DefaultConnectionHandler : ConnectionHandler {
    override fun connect(targetAddress: String, targetPort: Int): Pair<InputStream, OutputStream> {
        val socket = Socket(targetAddress, targetPort)
        return Pair(socket.getInputStream(), socket.getOutputStream())
    }
}

File: callers/deno/src/default-connection-handler.ts
/**
 * Базовая реализация ConnectionHandler через прямое TCP соединение.
 * Используется для тестирования SOCKS5 сервера.
 */

import type { ConnectionHandler } from "@src/connection-handler.ts";

/**
 * Реализация ConnectionHandler через прямое TCP соединение.
 */
export class DefaultConnectionHandler implements ConnectionHandler {
  async connect(
    targetAddress: string,
    targetPort: number,
  ): Promise<
    {
      reader: ReadableStreamDefaultReader<Uint8Array>;
      writer: WritableStreamDefaultWriter<Uint8Array>;
    }
  > {
    const conn = await Deno.connect({
      hostname: targetAddress,
      port: targetPort,
    });

    return {
      reader: conn.readable.getReader(),
      writer: conn.writable.getWriter(),
    };
  }
}
